plugins/chat/app/services/chat/stop_message_streaming.rb

```diff
@@ -33,12 +34,16 @@ def fetch_message(contract:)
       ::Chat::Message.find_by(id: contract.message_id)
     end
 
-    def can_join_channel(guardian:, message:)
-      guardian.can_join_chat_channel?(message.chat_channel)
+    def enforce_membership(guardian:, message:)
+      message.chat_channel.add(guardian.user) if guardian.user.bot?
+    end
+
+    def fetch_membership(guardian:, message:)
+      message.chat_channel.membership_for(guardian.user)
     end
 
     def can_stop_streaming(guardian:, message:)
-      guardian.is_admin? || message.user.id == guardian.user.id ||
+      guardian.user.bot? || guardian.is_admin? || message.user.id == guardian.user.id ||
```

Comment:
Do we need an `is_bot?` on the `guardian`?

Reply:
Yes not sure, I don't really like having the same info at two places, but on the other hand we have `guardian.admin` so that would make sense, but admin is related to permissions bot is not, it's more like a state of the user. not sure 🤷‍♂️